### ISSUE TITLE
Cargo check errors not reported when the error is on a curly brace

### ISSUE THREAD

**Issue by dNetGuru** [intellij-rust/intellij-rust#6689]
## Environment

* **IntelliJ Rust plugin version:** latest stable (0.3.138.3572-203)
* **Rust toolchain version:** rustc 1.49.0 (e1884a8e3 2020-12-29)
* **IDE name and version:** CLion Build #CL-203.6682.181, built on December 30, 2020
* **Operating system:** Windows 10 Build 19042.330

## Problem description
I use the external linter option to run `cargo check` on the fly due to the fact that the current static analyzer leaves a bit to be desired. I noticed at least one instance where an error emitted by `cargo check` is not reflected in the IDE.

## Steps to reproduce
1) Create a new Rust project in IntelliJ
2) `Ctrl+Alt+S` to open settings, then go to `Languages & Frameworks ->Rust -> Cargo` and 
-) make sure `External Linter` is set to `Cargo Check`
-) `Run external linter to analyze code on the fly` is checked
3) Copy and paste the code from the block below into the editor
```rust
async fn this_wont_work() {
    this_wont_work().await
}
```
4) Notice that a green check-mark is shown in the upper right corner of the IDE window and no part of the code is underlined
5) Run `cargo check` from the terminal window to see the error emitted by cargo check

![intellij-rust](https://user-images.githubusercontent.com/372719/104869460-f6c36180-58fa-11eb-8c75-eb85542e18ba.JPG)

I do suspect that this might have to do with the fact that the error is pointing to the "{" character since changing the return type of the `this_wont_work` function will result in the error being reported in the IDE:

![works](https://user-images.githubusercontent.com/372719/104869684-7d783e80-58fb-11eb-9523-d96b083af828.png)
